pkg/controller/storagecluster/storagecluster.go

```diff
@@ -1025,36 +1024,7 @@ func (c *Controller) nodeShouldRunStoragePod(
 		return false, true, nil
 	}
 
-	newPod, err := c.newSimulationPod(cluster, node.Name)
-	if err != nil {
-		logrus.Debugf("Failed to create a pod spec for node %v: %v", node.Name, err)
-		return false, false, err
-	}
-
-	taints := node.Spec.Taints
-	fitsNodeName, fitsNodeAffinity, fitsTaints := checkPredicates(newPod, node, taints)
-	if !fitsNodeName || !fitsNodeAffinity {
-		logrus.WithFields(logrus.Fields{
-			"nodeName":         node.Name,
-			"fitsNodeName":     fitsNodeName,
-			"fitsNodeAffinity": fitsNodeAffinity,
-		}).Debug("pod does not fit")
-		return false, false, nil
-	}
-
-	if !fitsTaints {
-		// Scheduled storage pods should continue running if they tolerate NoExecute taint
-		shouldContinueRunning := v1helper.TolerationsTolerateTaintsWithFilter(newPod.Spec.Tolerations, taints, func(t *v1.Taint) bool {
-			return t.Effect == v1.TaintEffectNoExecute
-		})
-		logrus.WithFields(logrus.Fields{
-			"nodeName":              node.Name,
-			"shouldContinueRunning": shouldContinueRunning,
-		}).Debug("pod does not fit taints")
-		return false, shouldContinueRunning, nil
-	}
-
-	return true, true, nil
+	return pxutil.CheckPredicates(node, cluster, c.StorageClusterSelectorLabels(cluster))
```

Comment:
Can we move this to `pkg/util` instead of portworx util? Doesn't feel good to refer driver specific utilities from the generic controller.

Reply:
it may cause circular reference, I think pkg can refer to driver, but driver could not refer to pkg.

Reply:
the change will import util/k8s, put it under util causes circular import, tried this.